File: frontend/src/types/support.ts
export type TicketStatus = "open" | "in-progress" | "resolved";

export interface SupportTicket {
  id: string;
  subject: string;
  description: string;
  screenshot_url: string | null;
  status: TicketStatus;
  response: string | null;
  responded_by: string | null;
  responded_at: string | null;
  submitted_by: string;
  created_at: string;
  updated_at: string;
}

export interface PaginatedTickets {
  items: SupportTicket[];
  total: number;
  page: number;
  limit: number;
}

File: backend/app/models/support.py
from typing import Literal

from pydantic import BaseModel, Field

TicketStatus = Literal["open", "in-progress", "resolved"]


class TicketCreate(BaseModel):
    subject: str = Field(min_length=1, max_length=200)
    description: str = Field(min_length=1)
    screenshot_url: str | None = None


class TicketUpdate(BaseModel):
    status: TicketStatus | None = None
    response: str | None = None


class TicketResponse(BaseModel):
    id: str
    subject: str
    description: str
    screenshot_url: str | None = None
    status: str
    response: str | None = None
    responded_by: str | None = None
    responded_at: str | None = None
    submitted_by: str
    created_at: str
    updated_at: str


class PaginatedTickets(BaseModel):
    items: list[TicketResponse]
    total: int
    page: int
    limit: int